### PR TITLE
Design and Implement Category Page

### PR DESCRIPTION
**Pull request by 0duaht**
Why?
Customers should be able to view a list of all the products in a particular category.

How?
Category page is populated with a list of products in each of the sub-categories nested in that particular category. Category page is paginated using the will_paginate gem.

https://www.pivotaltracker.com/story/show/113263227

### CODE REVIEW

**Comment by ChigboIO** [comment]
:+1: @andela-toduah 

@andela-oosiname It could be from your system. It is displaying well on my end.
Please check to see if you are using #Safari. It was the cause of our a similar problem we had. :smiley:

File: app/controllers/categories_controller.rb [hunk 1]
```diff
@@ -0,0 +1,18 @@
+class CategoriesController < ApplicationController
+  DEFAULT_LIMIT = 16
```
Comment (ChigboIO) [inline app/controllers/categories_controller.rb:2]: I think we didn't get to use this constant. Do we still need it :question: :grey_question:  :question:

**Comment by sofiengwin** [comment]
Nice work, I like the fact that you added breadcrumbs. I think it is something we should consider adding to the entire site.

File: app/controllers/categories_controller.rb (hunk 1 above)
Comment (0duaht) [inline app/controllers/categories_controller.rb:2]: Planned to, but didn't. Will be refactoring to use. Thanks for spotting that.

**Comment by 0duaht** [comment]
Thanks. I agree. :100:

**Comment by sofiengwin** [comment]
:+1:

**Comment by ChigboIO** [comment]
:+1:

**Comment by oluosiname** [comment]
:+1:

**Comment by ydaniju** [comment]
:+1:

**Comment by makinwab** [comment]
Nice work @andela-toduah. :+1:. I think it should also be possible for us to click on the `CATEGORIES` link and then it takes us to products in all categories.

File: app/controllers/categories_controller.rb [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+  def show
+    return category_not_present unless category
+    category_products = category.products
+    @products = category_products.paginate(
+      page: params[:page], per_page: DEFAULT_LIMIT)
```
Comment (makinwab) [inline app/controllers/categories_controller.rb:7]: **Nitpick - Code Quality**
Remember that when breaking an argument list, each argument should be placed on its own line.

For example

```
some_method(
      i_am_a_long_variable_name_that_will_never_fit_on_one_line_with_others,
      two,
      three
    )
```

**Pull request merged**